src/rosinstall_generator/generator.py

```diff
@@ -87,10 +87,12 @@ def _classify_repo_names(distro_name, repo_names):
     return names, unknown_names
 
 
-def _get_packages_for_repos(distro_name, repo_names, source=False):
+def _get_packages_for_repos(distro_name, repo_names, keywords, source=False):
```

Comment:
Instead of changing this function (which as a consequence would ignore one of the arguments) how about updating `repo_names` on the caller side instead?

Reply:
Handle `repo_names` on the caller side in 4bf324ad4fa40077f3be1a43cc963662a1db2133